src/lib.rs

```diff
@@ -483,16 +522,105 @@ impl Bump {
 
         Ok(Bump {
             current_chunk_footer: Cell::new(chunk_footer),
+            allocation_limit: Cell::new(None),
         })
     }
 
+    /// The allocation limit for this arena in bytes.
+    ///
+    /// ## Example
+    ///
+    /// ```
+    /// let bump = bumpalo::Bump::with_capacity(0);
+    ///
+    /// assert_eq!(bump.allocation_limit(), None);
+    ///
+    /// bump.set_allocation_limit(6);
+    ///
+    /// assert_eq!(bump.allocation_limit(), Some(6));
+    ///
+    /// bump.remove_allocation_limit();
+    ///
+    /// assert_eq!(bump.allocation_limit(), None);
+    /// ```
+    pub fn allocation_limit(&self) -> Option<usize> {
+        self.allocation_limit.get()
+    }
+
+    /// Set the allocation limit in bytes for this arena.
+    ///
+    /// Changing the limit for a `Bump` while it has live allocations does
+    /// nothing until a new allocation is attempted, at which point it may fail
+    /// due to the new limit.
+    ///
+    /// ## Example
+    ///
+    /// ```
+    /// let bump = bumpalo::Bump::with_capacity(0);
+    ///
+    /// bump.set_allocation_limit(0);
+    ///
+    /// assert!(bump.try_alloc(5).is_err());
+    /// ```
+    pub fn set_allocation_limit(&self, new_limit: usize) {
+        self.allocation_limit.set(Some(new_limit))
+    }
+
+    /// Remove the allocation limit in bytes for this arena.
+    ///
+    /// ## Example
+    ///
+    /// ```
+    /// let bump = bumpalo::Bump::with_capacity(0);
+    ///
+    /// bump.set_allocation_limit(0);
+    ///
+    /// assert!(bump.try_alloc(5).is_err());
+    ///
+    /// bump.remove_allocation_limit();
+    ///
+    /// assert!(bump.try_alloc(5).is_ok());
+    /// ```
+    pub fn remove_allocation_limit(&self) {
+        self.allocation_limit.set(None)
+    }
+
+    /// How much headroom an arena has before it hits its allocation
+    /// limit.
+    #[inline(always)]
```

Comment:
I don't think this doesn't need to be marked `inline` at all, let alone `inline(always)`. It is a method that is private to this compilation unit so LLVM will have no trouble looking at all of its callers and inlining as appropriate.

Reply:
Removed in latest commit